overlord/configstate/config/helpers.go

```diff
@@ -119,6 +120,90 @@ func GetFromChange(snapName string, subkeys []string, pos int, config map[string
 	return GetFromChange(snapName, subkeys, pos+1, configm, result)
 }
 
+// StoreConfigurationSnapshotMaybe makes a copy of config -> snapSnape configuration into the versioned config.
+// It doesn't do anything if there is no configuration for given snap in the state.
+func StoreConfigurationSnapshotMaybe(st *state.State, snapName string, rev snap.Revision) error {
+	var config map[string]interface{}                     // snap => configuration
+	var configSnapshots map[string]map[string]interface{} // snap => revision => configuration
```

Comment:
This seems to be more problematic than it should be... It seems that I'd need a custom json marshall/unmarshall code for this, otherwise I'm getting this:
json: unsupported type: map[snap.Revision]interface {} 
AFAIR the unmarshaller would need to use string for key anyway, as apparently (yeah, I was surprised by this, but see json.org) keys need to be strings (int is not supported).

So, to sump up, I'm not sure it's really worth complicating it.